frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber.tsx

```diff
@@ -0,0 +1,193 @@
+
+import { createBaseValidationRules } from '~utils/fieldValidation'
+import { SingleSelect } from '~components/Dropdown'
+import FormErrorMessage from '~components/FormControl/FormErrorMessage'
+import FormLabel from '~components/FormControl/FormLabel'
+import Input from '~components/Input'
+import NumberInput from '~components/NumberInput'
+import Textarea from '~components/Textarea'
+import Toggle from '~components/Toggle'
+
+import { DrawerContentContainer } from './common/DrawerContentContainer'
+import { FormFieldDrawerActions } from './common/FormFieldDrawerActions'
+import { EditFieldProps } from './common/types'
+import { useEditFieldForm } from './common/useEditFieldForm'
+
+type EditNumberProps = EditFieldProps<NumberFieldBase>
+
+const EDIT_NUMBER_FIELD_KEYS = ['title', 'description', 'required'] as const
+
+type EditNumberInputs = Pick<
+  NumberFieldBase,
+  typeof EDIT_NUMBER_FIELD_KEYS[number]
+> & {
+  ValidationOptions: {
+    selectedValidation: NumberSelectedValidation | ''
+    customVal: number | ''
+  }
+}
+
+const transformNumberFieldToEditForm = (
+  field: NumberFieldBase,
+): EditNumberInputs => {
+  const nextValidationOptions =
+    field.ValidationOptions.selectedValidation === null
+      ? {
+          selectedValidation: '' as const,
+          customVal: '' as const,
+        }
+      : {
+          selectedValidation: field.ValidationOptions.selectedValidation,
+          customVal:
+            field.ValidationOptions.customVal === null
+              ? ('' as const)
+              : field.ValidationOptions.customVal,
+        }
+  return {
+    ...pick(field, EDIT_NUMBER_FIELD_KEYS),
+    ValidationOptions: nextValidationOptions,
+  }
+}
+
+const transformNumberEditFormToField = (
+  form: EditNumberInputs,
+  originalField: NumberFieldBase,
+): NumberFieldBase => {
+  const nextValidationOptions =
+    form.ValidationOptions.selectedValidation === ''
+      ? {
+          selectedValidation: null,
+          customVal: null,
+        }
+      : form.ValidationOptions
+  return extend({}, originalField, form, {
+    ValidationOptions: nextValidationOptions,
+  })
+}
+
+export const EditNumber = ({ field }: EditNumberProps): JSX.Element => {
+  const {
+    register,
+    formState: { errors },
+    isSaveEnabled,
+    buttonText,
+    handleUpdateField,
+    watch,
+    control,
+    clearErrors,
+    isLoading,
+    handleCancel,
+    setValue,
+  } = useEditFieldForm<EditNumberInputs, NumberFieldBase>({
+    field,
+    transform: {
+      input: transformNumberFieldToEditForm,
+      output: transformNumberEditFormToField,
+    },
+  })
+
+  const requiredValidationRule = useMemo(
+    () => createBaseValidationRules({ required: true }),
+    [],
+  )
+
+  const watchedSelectedValidation = watch(
+    'ValidationOptions.selectedValidation',
+  )
+
+  const customValValidationOptions: RegisterOptions = useMemo(
+    () => ({
+      required: {
+        value: watchedSelectedValidation !== '',
```

Comment:
nit+perf: it's recommended to use `getValues` inside register options: https://react-hook-form.com/api/useform/getvalues

this means this memoized object will not recreate on selected validation change